src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -58,44 +72,116 @@ public override async Task<LSP.CompletionItem[]> HandleRequestAsync(LSP.Completi
                 .WithChangedOption(CompletionServiceOptions.DisallowAddingImports, true);
 
             var completionService = document.Project.LanguageServices.GetRequiredService<CompletionService>();
-            var list = await completionService.GetCompletionsAsync(document, position, options: completionOptions, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+            // TO-DO: More LSP.CompletionTriggerKind mappings are required to properly map to Roslyn CompletionTriggerKinds.
+            // https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1178726
+            var triggerCharacter = char.Parse(request.Context.TriggerCharacter);
+            var triggerKind = ProtocolConversions.LSPToRoslynCompletionTriggerKind(request.Context.TriggerKind);
+            var completionTrigger = new CompletionTrigger(triggerKind, triggerCharacter);
+
+            var list = await completionService.GetCompletionsAsync(document, position, completionTrigger, options: completionOptions, cancellationToken: cancellationToken).ConfigureAwait(false);
             if (list == null)
             {
                 return Array.Empty<LSP.CompletionItem>();
             }
 
             var lspVSClientCapability = context.ClientCapabilities?.HasVisualStudioLspCapability() == true;
 
-            return list.Items.Select(item => CreateLSPCompletionItem(request, item, lspVSClientCapability)).ToArray();
+            return list.Items.Select(item => CreateLSPCompletionItem(request, item, lspVSClientCapability, completionTrigger)).ToArray();
 
             // local functions
-            static LSP.CompletionItem CreateLSPCompletionItem(LSP.CompletionParams request, CompletionItem item, bool useVSCompletionItem)
+            static LSP.CompletionItem CreateLSPCompletionItem(
+                LSP.CompletionParams request,
+                CompletionItem item,
+                bool useVSCompletionItem,
+                CompletionTrigger completionTrigger)
             {
                 if (useVSCompletionItem)
                 {
-                    var vsCompletionItem = CreateCompletionItem<LSP.VSCompletionItem>(request, item);
+                    var vsCompletionItem = CreateCompletionItem<LSP.VSCompletionItem>(request, item, completionTrigger);
                     vsCompletionItem.Icon = new ImageElement(item.Tags.GetFirstGlyph().GetImageId());
                     return vsCompletionItem;
                 }
                 else
                 {
-                    var roslynCompletionItem = CreateCompletionItem<RoslynCompletionItem>(request, item);
+                    var roslynCompletionItem = CreateCompletionItem<RoslynCompletionItem>(request, item, completionTrigger);
                     roslynCompletionItem.Tags = item.Tags.ToArray();
```

Comment:
I think roslyn completion item can be removed - it was used in the old liveshare implementation.  If vscompletionItem is not supported, we should probably just return a regular CompletionItem